usgo/gocongress
# Tournament signup

This is a relatively substantial branch -- it adds:

* The ability for tournaments to be labeled as online, and to designate on which server they will take place.
* The ability for tournaments to be made available for registration during registration.
* The ability for attendees to enter their server usernames for online tournaments as part of registration, and to have front-end JavaScript check whether those usernames exist to reduce incorrect usernames.
* There is also a significant re-factoring and improvement of the registration fields to allow for better display of errors and required fields.

TODO: Add testing to cover new tournament / registration functionality.

TO TEST:

* Pull the branch and run db:migrate.
* As an admin, create some new tournaments, or edit existing ones, and make some that are online and/or can be registered for as part of registration.
* With a new or existing attendee, check the registration page to verify that you can now view and register for tournaments.
* Verify that if you register for an online tournament but do not supply a username, you get an error after clicking "submit".
* Verify that when you type in a username for a server, it successfully checks whether it exists.
* Verify that if there are multiple tournaments for the same server, and therefore multiple username inputs for the same username, when you type into one of them it synchronizes any others with it.
